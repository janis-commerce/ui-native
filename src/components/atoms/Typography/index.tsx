import React, {ReactElement} from 'react';
import {Text, StyleProp, TextStyle, TextProps} from 'react-native';
import getStyleByTypography from './utils/getStyleByTypography';

export type TypographyType = 'display' | 'heading' | 'title' | 'label' | 'body' | 'overline';

export type TypographySize = 'large' | 'medium' | 'small';

interface TypographyProps extends TextProps {
	children?: ReactElement | string;
	style?: StyleProp<TextStyle>;
	type?: TypographyType;
	size?: TypographySize;
}

const Typography = ({children, style, type, size, ...props}: TypographyProps) => {
	if (!children) {
		return null;
	}

	const typographyStyles = getStyleByTypography(
		type as TypographyType | 'string',
		size as TypographySize | 'string'
	);

	return (
		<Text style={[style, typographyStyles.typography]} {...props}>
			{children}
		</Text>
	);
};

export default Typography;
