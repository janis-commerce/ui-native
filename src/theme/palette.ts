import {
	Alert,
	Base,
	Black,
	Env,
	Error,
	GreyScale,
	Primary,
	Success,
	Warning,
	White,
} from '../ts/interfaces/colors';

const primary: Primary = {
	main: '#2979FF',
	dark: '#1759FF',
	light: '#02BFFB',
};
const black: Black = {
	main: '#2F2F2F',
	dark: '#050505',
};
const white: White = {
	main: '#E8EAF6',
	dark: '#D0D3E3',
	light: '#F4F5FB',
};
const grey: GreyScale = {
	100: '#DDDFE2',
	200: '#D5D7DB',
	300: '#C4C6CC',
	400: '#A8AAAC',
	500: '#939598',
	600: '#747679',
	700: '#585858',
};
const base: Base = {
	white: '#fff',
	black: '#000',
};
const success: Success = {
	main: '#1DB779',
	dark: '#109D59',
};
const error: Error = {
	main: '#FF4343',
	dark: '#FF2A2A',
};
const warning: Warning = {
	main: '#FF8D10',
	dark: '#FF6E08',
};
const alert: Alert = {
	main: '#FFCE17',
	dark: '#FFBA0C',
};
const environment: Env = {
	qa: '#1DB779',
	beta: '#F13B70',
};

export {primary, black, white, grey, base, success, error, warning, alert, environment};
