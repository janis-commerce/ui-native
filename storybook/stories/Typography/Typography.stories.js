import React from 'react';
import Typography from 'atoms/Typography';
import typography from 'theme/typography';

const types = Array.from(Object.keys(typography));

const sizes = ['large', 'medium', 'small', 'default'];

export default {
	title: 'Components/Typography',
	argTypes: {
		type: {
			options: types,
			control: {type: 'select'},
		},
		size: {
			options: sizes,
			control: {type: 'select'},
		},
		color: {
			control: {type: 'color'},
		},
	},
};

export const DefaultProps = ({textToDisplay, color, type, size}) => (
	<Typography color={color} type={type} size={size}>
		{textToDisplay}
	</Typography>
);

DefaultProps.storyName = 'default props';

DefaultProps.args = {
	textToDisplay: 'Hola',
	color: '#000000',
	type: 'display',
	size: 'medium',
};
