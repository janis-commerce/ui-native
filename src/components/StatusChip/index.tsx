import React, {ReactElement, isValidElement} from 'react';
import {StyleSheet, Text, ViewProps, View} from 'react-native';
import {base, grey, primary} from '../../theme/palette';

interface StatusChipProps extends ViewProps {
	children?: ReactElement | string;
	background?: string;
}

const StatusChip = ({children, ...props}: StatusChipProps) => {
	const isChildrenAString = typeof children === 'string';
	const isCustomComponent = isValidElement(children);

	const hasToRenderStatusChip = !children || (!isChildrenAString && !isCustomComponent);

	if (hasToRenderStatusChip) {
		return null;
	}

	return (
		<View style={styles(props).ViewStyles} {...props}>
			{isCustomComponent ? children : <Text style={styles(props).TextStyles}>{children}</Text>}
		</View>
	);
};

const styles = ({background}: StatusChipProps) =>
	StyleSheet.create({
		ViewStyles: {
			height: 24,
			flexDirection: 'row',
			alignItems: 'center',
			paddingLeft: 12,
			paddingRight: 12,
			borderRadius: 12,
			backgroundColor: background ?? base.white,
			borderWidth: 1,
			borderColor: background ?? grey['300'],
		},
		TextStyles: {
			fontSize: 15,
			lineHeight: 18,
			fontFamily: 'Roboto',
			fontWeight: '900',
			textAlign: 'center',
			color: background ? base.white : primary.main,
		},
	});

export default StatusChip;
