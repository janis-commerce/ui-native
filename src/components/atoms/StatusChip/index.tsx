import React, {ReactElement, isValidElement} from 'react';
import {StyleSheet, ViewProps, View} from 'react-native';
import {base, grey, primary} from 'theme/palette';
import Text from 'atoms/Text';
import {horizontalScale, moderateScale, scaledForDevice} from 'scale';

interface StatusChipProps extends ViewProps {
	children?: ReactElement | string;
	background?: string;
}

const validHeight = scaledForDevice(24, moderateScale);
const validPadding = scaledForDevice(12, horizontalScale);
const validBorderRadius = scaledForDevice(12, moderateScale);
const validBorderWidth = scaledForDevice(1, moderateScale);
const validFontSize = scaledForDevice(13, moderateScale);
const validLineHeight = scaledForDevice(18, moderateScale);

const styles = ({background}: StatusChipProps) =>
	StyleSheet.create({
		ViewStyles: {
			height: validHeight,
			flexDirection: 'row',
			alignItems: 'center',
			paddingLeft: validPadding,
			paddingRight: validPadding,
			borderRadius: validBorderRadius,
			backgroundColor: background ?? base.white,
			borderWidth: validBorderWidth,
			borderColor: background ?? grey['300'],
		},
		TextStyles: {
			fontSize: validFontSize,
			lineHeight: validLineHeight,
			fontFamily: 'Roboto',
			fontWeight: '900',
			textAlign: 'center',
			color: background ? base.white : primary.main,
		},
	});

const StatusChip = ({children, ...props}: StatusChipProps) => {
	const isChildrenAString = typeof children === 'string';
	const isCustomComponent = isValidElement(children);

	const hasToRenderStatusChip = !children || (!isChildrenAString && !isCustomComponent);

	if (hasToRenderStatusChip) {
		return null;
	}

	return (
		<View style={styles(props).ViewStyles} {...props}>
			{isCustomComponent ? children : <Text style={styles(props).TextStyles}>{children}</Text>}
		</View>
	);
};

export default StatusChip;
