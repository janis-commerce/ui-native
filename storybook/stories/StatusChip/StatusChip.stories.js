import StatusChip from 'atoms/StatusChip';
import Text from 'atoms/Text';
import React from 'react';

export default {
	title: 'Components/StatusChip',
	argTypes: {
		background: {
			control: {type: 'color'},
		},
	},
};

export const WithText = (props) => <StatusChip {...props}>Partially delivered</StatusChip>;

WithText.storyName = 'With text';

WithText.args = {
	background: '#2979FF',
};
const TextStyles = {
	color: '#fff',
	fontSize: 13,
	lineHeight: 18,
	fontFamily: 'Roboto',
	fontWeight: '900',
	textAlign: 'center',
};

export const WithCustomComponent = (props) => (
	<StatusChip {...props}>
		<Text style={TextStyles}>Delivered</Text>
	</StatusChip>
);

WithCustomComponent.storyName = 'With custom component';

WithCustomComponent.args = {
	background: '#2979FF',
};
