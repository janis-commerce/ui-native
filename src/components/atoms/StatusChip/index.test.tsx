import 'react-native';
import React from 'react';
import {create} from 'react-test-renderer';
import StatusChip from './index';
import Typography from 'atoms/Typography';
import palette from 'theme/palette';

describe('StatusChip component', () => {
	describe('return error because is not a valid children', () => {
		it('undefined children', () => {
			const {toJSON} = create(<StatusChip />);
			expect(toJSON()).toBeNull();
		});
	});

	describe('render correct', () => {
		it('text is passed', () => {
			const {toJSON} = create(<StatusChip>Partially delivered</StatusChip>);
			expect(toJSON()).toMatchSnapshot();
		});

		it('if a background is passed', () => {
			const {toJSON} = create(
				<StatusChip background={palette.primary.blue.normal}>Delivered</StatusChip>
			);
			expect(toJSON()).toMatchSnapshot();
		});

		it('a valid component is passed', () => {
			const {toJSON} = create(
				<StatusChip>
					<Typography>Custom component</Typography>
				</StatusChip>
			);
			expect(toJSON()).toMatchSnapshot();
		});
	});
});
