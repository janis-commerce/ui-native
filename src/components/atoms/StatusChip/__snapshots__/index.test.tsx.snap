// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusChip component render correct a valid component is passed 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#C4C6CC",
      "borderRadius": 25,
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 50,
      "paddingLeft": 25,
      "paddingRight": 25,
    }
  }
>
  <Text
    style={
      Array [
        undefined,
        Object {
          "color": "#000000",
          "fontSize": 29,
          "fontWeight": "400",
          "lineHeight": 18,
        },
      ]
    }
  >
    Custom component
  </Text>
</View>
`;

exports[`StatusChip component render correct if a background is passed 1`] = `
<View
  background="#2979FF"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2979FF",
      "borderColor": "#2979FF",
      "borderRadius": 25,
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 50,
      "paddingLeft": 25,
      "paddingRight": 25,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 27,
          "fontWeight": "900",
          "lineHeight": 37.5,
          "textAlign": "center",
        },
        Object {
          "color": "#000000",
          "fontSize": 29,
          "fontWeight": "400",
          "lineHeight": 18,
        },
      ]
    }
  >
    Delivered
  </Text>
</View>
`;

exports[`StatusChip component render correct text is passed 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#C4C6CC",
      "borderRadius": 25,
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 50,
      "paddingLeft": 25,
      "paddingRight": 25,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#2979FF",
          "fontFamily": "Roboto",
          "fontSize": 27,
          "fontWeight": "900",
          "lineHeight": 37.5,
          "textAlign": "center",
        },
        Object {
          "color": "#000000",
          "fontSize": 29,
          "fontWeight": "400",
          "lineHeight": 18,
        },
      ]
    }
  >
    Partially delivered
  </Text>
</View>
`;
